### PR TITLE
Janiborg no longer starts with floor buffer and now can be toggled

### PR DESCRIPTION
**Pull request by Henri215**
## What Does This PR Do

Alternative version of #15864
Janiborgs no longer starts with a floor buffer, its now an upgrade module.
Floor buffer now can be toggled on/off and the movement speed is decreased while active.
## Why It's Good For The Game

Janiborgs are too good at cleaning, most of the time leaving poor organic janitors to stare while they just cleans the room by simply existing.
Janiborgs advanced mop wont be ignored as much with floor buffers no longer being something they start with.
By decreasing their speed while the buffer is active we would be mainly nerfing their combo with vtec, that did turn them into cleaning monsters before.

## Images of changes

WITHOUT VTEC
https://user-images.githubusercontent.com/77684085/211211566-fe9a7245-ba02-478b-857f-fb9f7463ece8.mp4

WITH VTEC
https://user-images.githubusercontent.com/77684085/211211549-b751fe96-e0ea-4914-85a1-0bbeae8a4a7d.mp4

## Testing

- See videos above
## Changelog
:cl:
tweak: Janiborgs no longer starts with floor buffers, its now an upgrade module
tweak: Floor buffer now can be toggled on/off and the movement speed is decreased while active
/:cl:

### CODE REVIEW

**Comment by Henri215** [comment]
If its considered too big of a nerf then the alternative would be to make it something they start with again.

**Comment by Christasmurf** [comment]
You could additionally/alternatively make the buffer drain power faster while active

**Comment by Henri215** [comment]
> You could additionally/alternatively make the buffer drain power faster while active

I did think of this while making the PR, its an option.

**Comment by procdrone** [comment]
this looks reasonable

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 1]
```diff
@@ -363,3 +363,38 @@
...
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "upgrade"
+	require_module = TRUE
+	module_type = /obj/item/robot_module/janitor
+	var/buffer_speed = 1 // How much speed the cyborg loses while the buffer is active
+	var/mob/living/silicon/robot/cyborg
+
+/obj/item/borg/upgrade/floorbuffer/do_install(mob/living/silicon/robot/R)
+	var/obj/item/borg/upgrade/floorbuffer/U = locate() in R
+	if(U)
```
Comment (S34NW) [inline code/game/objects/items/robot/robot_upgrades.dm:379]:
```suggestion
	for(var/obj/item/borg/upgrade/floorbuffer/U in R)
```

**Review by S34NW — APPROVED** [review]

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -363,3 +363,37 @@
...
 		deactivate()
+
+/obj/item/borg/upgrade/floorbuffer
+	name = "janitor cyborg floor buffer upgrade"
+	desc = "A floor buffer upgrade kit that can be attached to janitor cyborgs."
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "upgrade"
+	require_module = TRUE
+	module_type = /obj/item/robot_module/janitor
+	var/buffer_speed = 1 // How much speed the cyborg loses while the buffer is active
```
Comment (Sirryan2002) [inline code/game/objects/items/robot/robot_upgrades.dm:374]:
```suggestion
	/// How much speed the cyborg loses while the buffer is active
	var/buffer_speed = 1
```

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 3]
```diff
@@ -107,6 +107,7 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 	var/magpulse = FALSE
 	var/ionpulse = FALSE // Jetpack-like effect.
 	var/ionpulse_on = FALSE // Jetpack-like effect.
+	var/floorbuffer = FALSE //Does it clean the tile under it?
```
Comment (Sirryan2002) [inline code/modules/mob/living/silicon/robot/robot.dm:110]:
```suggestion
	/// Does it clean the tile under it?
	var/floorbuffer = FALSE
```

**Review by Sirryan2002 — APPROVED** [review]

**Pull request merged**